pandas/tests/arrays/test_array.py

```diff
@@ -393,6 +393,21 @@ def test_array_not_registered(registry_without_decimal):
     tm.assert_equal(result, expected)
 
 
+def test_datetime64_eq_string():
+    # Issue https://github.com/pandas-dev/pandas/issues/45506
+    # Catch OverflowError when comparing datetime64 and string
+    data = [
+        {"f_0": "2015-07-01", "f_2": "08335394550"},
+        {"f_0": "2015-07-02", "f_2": "+49 (0) 0345 300033"},
+        {"f_0": "2015-07-03", "f_2": "+49(0)2598 04457"},
+        {"f_0": "2015-07-04", "f_2": "0741470003"},
+        {"f_0": "2015-07-05", "f_2": "04181 83668"},
+    ]
+    dtypes = {"f_0": "datetime64[ns]", "f_2": "string"}
+    df = pd.DataFrame(data=data).astype(dtypes)
+    assert not df["f_0"].eq(df["f_2"]).any()
```

Comment:
construct the expected frame and use `tm.assert_frame_equal`

Reply:
`tm.assert_frame_equal` isn't working. After investigation I found out it was `tm.assert_series_equal`. Or you meant something else?